.editorconfig

```diff
@@ -298,3 +298,19 @@ dotnet_naming_style.end_in_async.required_prefix =
 dotnet_naming_style.end_in_async.required_suffix = Async
 dotnet_naming_style.end_in_async.capitalization = pascal_case
 dotnet_naming_style.end_in_async.word_separator = 
+
+# Exclude extension method 'this' parameter
```

Comment:
Would you speak to these new settings in more detail?

Reply:
Sure:
```dotnet_code_quality.CA1062.exclude_extension_method_this_parameter = true```
This removed the false positive when an extension method wasn't checking the `this` parameter. Otherwise we'd have had to suppress ALL of these instances and there were quite a few.

```
# CA2016: Forward the 'CancellationToken' parameter to methods that take one
dotnet_diagnostic.CA2016.severity = warning

# CA1805: Do not initialize unnecessarily
dotnet_diagnostic.CA1805.severity = warning

# CA1834: Consider using 'StringBuilder.Append(char)' when applicable
dotnet_diagnostic.CA1834.severity = warning

# CA2000: Dispose objects before losing scope
dotnet_diagnostic.CA2000.severity = warning
```
These were all **warnings** in the original package analyzers but has been upgraded to an **error**. I plan to take these in other PRs to remove the need for these lines to be configured and we'll put them back as errors.

Reply:
I suggest you add a comment for them to that effect.